### PR TITLE
Support for PackageVersionsRetention rule

### PR DESCRIPTION
**Pull request by aan11**
Changes request in the last pull request (https://github.com/NuGet/NuGet.Server/issues/17). Thank you.

### CODE REVIEW

**Comment by aan11** [comment]
I have add new test for this feature (pull request 18) and rename the property to PackageVersionRetention.

PS: I am a beginner with github
Thanks! :)

File: test/NuGet.Server.Tests/ServerPackageRepositoryTest.cs [hunk 1]
```diff
@@ -109,6 +110,41 @@ public void ServerPackageRepositoryRemovePackage()
...
+                    repository.AddPackage(CreatePackage("test", "1.8"));
+                    repository.AddPackage(CreatePackage("test", "1.9"));
+                    repository.AddPackage(CreatePackage("test", "2.0"));
+                    repository.AddPackage(CreatePackage("test", "2.1"));
+                    repository.AddPackage(CreatePackage("test", "2.2"));
+                }, null, getIntSetting);
+
+                // Act
+                serverRepository.AddPackage(CreatePackage("test", "2.3"));
+                var packages = serverRepository.GetPackages().OrderBy(p => p.Version).ToList();
```
Comment (xavierdecoster) [inline test/NuGet.Server.Tests/ServerPackageRepositoryTest.cs]: This has been adjusted to use `FindPackagesById` in the [implementation](https://github.com/NuGet/NuGet.Server/pull/18/files#diff-5fe4d2ce57f43bc716ec5562ea2063f8R271). The test should follow the same code path and use `FindPackagesById` too.

**Comment by skofman1** [comment]
Thanks for you contribution! One more comment and we are good to merge :)

**Comment by skofman1** [comment]
@emgarten please review as well

File: src/NuGet.Server/Infrastructure/ServerPackageRepository.cs [hunk 2]
```diff
@@ -685,6 +701,14 @@ private bool EnableDelisting
                 return _getSetting("enableDelisting", false);
             }
         }
+        private int PackageVersionRetention
```
Comment (shishirx34) [inline src/NuGet.Server/Infrastructure/ServerPackageRepository.cs]: nit: newline, below alignment too.

File: src/NuGet.Server/Infrastructure/ServerPackageRepository.cs [hunk 3]
```diff
@@ -702,7 +708,7 @@ private bool EnableDelisting
             }
         }
         private int PackageVersionRetention
-		{
+    {
```
Comment (xavierdecoster) [inline src/NuGet.Server/Infrastructure/ServerPackageRepository.cs]: nit: spacing seems off

File: test/NuGet.Server.Tests/ServerPackageRepositoryTest.cs (hunk 1 above)
Comment (xavierdecoster) [inline test/NuGet.Server.Tests/ServerPackageRepositoryTest.cs]: Looks like you forgot to adjust this one to use `FindPackagesById`?

Comment (aan11) [inline test/NuGet.Server.Tests/ServerPackageRepositoryTest.cs]: I have fix it.

**Comment by xavierdecoster** [comment]
Once the remaining comments addressed and @aan11 signed the CLA (see https://github.com/NuGet/NuGet.Server/pull/18#issuecomment-253542241), I think these changes are good to go.

**Comment by skofman1** [comment]
@aan11 , can you please sign off on the license agreement, so we can merge your PR?
You can do this here: https://cla2.dotnetfoundation.org.

**Comment by wiruzman** [comment]
This one is dead?

I am planning to do a retention job unless this one going to merge into the release.

**Comment by chenriksson** [comment]
@skofman1 @wiruzman It looks like the PR was never finished because the contributor license was never signed. Also, the PR is based off the old v2 version of NuGet.Server.

If the changes are still desired, feel free to open a new PR off of the latest (v3) and sign the license.

**Pull request closed without merging**